File: fixtures/scrapy_output/get_with_data2.py
from scrapy import Request, Spider


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        headers = {
            'X-Api-Key': '{admin_api_key}',
            'Content-Type': 'application/json',
        }
        yield Request(
            'https://api.newrelic.com/v2/alerts_policy_channels.json?policy_id=policy_id&channel_ids=channel_id',
            callback=self.parse,
            method='PUT',
            headers=headers,
        )

    def parse(self, response):
        pass

File: test/fixtures/javascript-xhr/get_with_data2.js
let xhr = new XMLHttpRequest();
xhr.withCredentials = true;
xhr.open('PUT', 'http://localhost:28139/v2/alerts_policy_channels.json?policy_id=policy_id&channel_ids=channel_id');
xhr.setRequestHeader('X-Api-Key', '{admin_api_key}');
xhr.setRequestHeader('Content-Type', 'application/json');

xhr.onload = function() {
  console.log(xhr.response);
};

xhr.send();
